### PR TITLE
Add & remove points & segments by id

### PR DESCRIPTION
**Pull request by jdelStrother**
We've found it useful to be able to refer to points & segments by their ID.  How about something like this, based upon @chainlink's work in https://github.com/bbcrd/peaks.js/pull/64 ?

### CODE REVIEW

**Comment by jdelStrother** [comment]
^ I'll look into those test failures, not sure why I didn't see those locally

**Comment by jdelStrother** [comment]
Should be fixed now - I think the current failures are just sporadic ones that are fixed by #142

**Comment by chrisn** [comment]
Hi Jonathan, thanks for these changes and for fixing the test failures. We'll take a look and merge soon. (Tests aren't passing locally for me on master at the moment).

**Comment by jdelStrother** [comment]
Sure.   Let me know if you'd like to take a closer look at any failing tests - `npm run build && npm test -- --browsers Firefox` works ok for me (which is pretty much what Travis does), but I ran into some failures with Chrome - there's some annoying weirdness with currentTime not getting updated after seeking.

File: src/main/markers/waveform.segments.js [hunk 1]
```diff
@@ -25,9 +26,9 @@ define([
       return view;
     });
 
-    var createSegmentWaveform = function (segmentId, startTime, endTime, editable, color, labelText) {
+    var createSegmentWaveform = function (id, startTime, endTime, editable, color, labelText) {
```
Comment (chrisn) [inline src/main/markers/waveform.segments.js]: I recommend keeping parameters in the same order between this and `createSegment`, and `addSegment` in main.js.

File: src/main.js [hunk 2]
```diff
@@ -281,12 +281,13 @@ define('peaks', [
       get: function () {
         var self = this;
 
-        function addSegment (startTime, endTime, editable, color, labelText) {
+        function addSegment (startTime, endTime, editable, color, labelText, id) {
```
Comment (chrisn) [inline src/main.js]: The number of parameters is becoming unwieldy, so I'd prefer we address issue #6 before (or as part of) making this change.

Comment (jdelStrother) [inline src/main.js]: I can hopefully look at that sometime next week.   How about merging #142 to fix Travis?

Comment (chrisn) [inline src/main.js]: Done! That'd be great if you're happy to take a look at #6, otherwise I can try to find time.

**Comment by jdelStrother** [comment]
I've rebased this on top of the latest master, it's a lot cleaner with the recent changes to the addPoint/addSegment method signatures.

File: src/main/markers/waveform.segments.js [hunk 3]
```diff
@@ -25,10 +25,10 @@ define([
       return view;
     });
 
+    var segmentId = 0;
     var createSegmentWaveform = function (options) {
-      var segmentId = "segment" + self.segments.length;
       var segment = {
-        id: segmentId,
+        id: options.id || ('peaks.segment.' + segmentId++),
```
Comment (chrisn) [inline src/main/markers/waveform.segments.js]: A minor nitpick, but I think we should allow zero as a valid ID, and only set a default ID if undefined or null.

Comment (jdelStrother) [inline src/main/markers/waveform.segments.js]: I could - we'd need to update waveform-data as well though, which ends up replacing 0`s at https://github.com/bbcrd/waveform-data.js/blob/master/lib/core.js#L194-L195

Comment (chrisn) [inline src/main/markers/waveform.segments.js]: I see the `set_point` method in waveform-data handles this correctly, but `set_segment` is inconsistent. I'll fix and update it.

Comment (chrisn) [inline src/main/markers/waveform.segments.js]: I've just updated waveform-data.js to v1.5.2, which fixes this issue. I can't publish a new version to npm as this needs @oncletom, who's currently away, so for now I've added a git URL to package.json.

Comment (jdelStrother) [inline src/main/markers/waveform.segments.js]: Looks good, I've updated the PR with support for 0-ids.

**Pull request merged**